src/Microsoft.TemplateEngine.Core/Operations/InlineMarkupConditional.cs

```diff
@@ -0,0 +1,233 @@
+
+        public bool WholeLine { get; }
+
+        public IOperation GetOperation(Encoding encoding, IProcessorState processorState)
+        {
+            TokenTrie structureTrie = new TokenTrie();
+            TokenTrie closeConditionTrie = new TokenTrie();
+            TokenTrie scanBackTrie = new TokenTrie();
+
+            byte[] openOpenElementTokenBytes = processorState.Encoding.GetBytes(Tokens.OpenOpenElementToken);
+            scanBackTrie.AddToken(openOpenElementTokenBytes);
+            int openOpenElementToken = structureTrie.AddToken(openOpenElementTokenBytes);
+            int openCloseElementToken = structureTrie.AddToken(processorState.Encoding.GetBytes(Tokens.OpenCloseElementToken));
+            int closeCloseElementToken = structureTrie.AddToken(processorState.Encoding.GetBytes(Tokens.CloseElementTagToken));
+
+            int selfClosingElementEndToken = -1;
+            if (Tokens.SelfClosingElementEndToken != null)
+            {
+                selfClosingElementEndToken = structureTrie.AddToken(processorState.Encoding.GetBytes(Tokens.SelfClosingElementEndToken));
+            }
+
+            closeConditionTrie.AddToken(processorState.Encoding.GetBytes(Tokens.CloseConditionExpression));
+            MarkupTokenMapping mapping = new MarkupTokenMapping(
+                openOpenElementToken,
+                openCloseElementToken,
+                closeCloseElementToken,
+                selfClosingElementEndToken
+            );
+
+            IReadOnlyList<byte[]> start = new[] { processorState.Encoding.GetBytes(Tokens.OpenConditionExpression) };
+            return new Impl(this, start, structureTrie, closeConditionTrie, scanBackTrie, mapping, _id);
+        }
+
+        public class Impl : IOperation
+        {
+            private readonly TokenTrie _closeConditionTrie;
+            private readonly InlineMarkupConditional _definition;
+            private readonly MarkupTokenMapping _mapping;
+            private readonly TokenTrie _scanBackTrie;
+            private readonly TokenTrie _structureTrie;
+
+            public Impl(InlineMarkupConditional definition, IReadOnlyList<byte[]> tokens, TokenTrie structureTrie, TokenTrie closeConditionTrie, TokenTrie scanBackTrie, MarkupTokenMapping mapping, string id)
+            {
+                _definition = definition;
+                Id = id;
+                Tokens = tokens;
+                _mapping = mapping;
+                _structureTrie = structureTrie;
+                _scanBackTrie = scanBackTrie;
+                _closeConditionTrie = closeConditionTrie;
+            }
+
+            public string Id { get; }
+
+            public IReadOnlyList<byte[]> Tokens { get; }
+
+            public int HandleMatch(IProcessorState processor, int bufferLength, ref int currentBufferPosition, int token, Stream target)
+            {
+                bool flag;
+                if (processor.Config.Flags.TryGetValue("conditionals", out flag) && !flag)
```

Comment:
"conditionals" is not the correct flag here. We Need a new one for this operation.
Actually, the checking of these flags was partly moved to ProcessorState.Run(), but apparently the checking was not completely removed from the operations.